### ISSUE TITLE
SIGSEGV crashes on seemingly innocuous commands

### ISSUE THREAD

**Issue by AutumnMux** [brazilofmux/tinymux#700]
* OS is Ubuntu 22.04.1
* TinyMUX version is 2.13.0.4 

Over the past week I've seen three SIGSEGV crashes, all on separate commands:

MUX-20240102-174505.log:2024.0304:184850 MUX BUG/INFO : Command: ‘< process_output >’

MUX-20240304-185205.log:2024.0304:200707 MUX BUG/INFO : Command: ‘look’

MUX-20240304-200707.log:2024.0306:174653 MUX BUG/INFO : Command: ‘WHO’

These crashes are not repeatable in any obvious way.  No coredumps were generated. 

Are there any specific log events that would be particularly helpful to log?  I've done @admin log=debug, but have not had a crash since doing so.

If there's other information that would be helpful, please let me know.

**Comment by brazilofmux** [comment]
Depending on your appetite for problems (you are running an Alpha version), you can either go back to TinyMUX 2.12.0.10 or run different alpha versions of TinyMUX 2.13 (2.13.0.0 through 2.13.0.4) in an attempt to bracket the issue.

When switching versions, be sure to rebuild dependencies with 'make depend' and 'make realclean;make'.

The latest changes in networking try to fix other long-standing blocking issues with the SSL code, but the new changes are an experimental approach to non-blocking code. I've seen three SIGSEGVs in '< process_output_ssl >', but I've been running with the SSL turned on. I imagine there are problems with process_ouput as well.

In summary, it is unfortunate, but the issue is probably from some pretty risky changes in the networking code.

If you do bracket the intermitant problem to a specific version (happens on one minor version but not on the previous one), please feel free to share that information.

**Comment by AutumnMux** [comment]
Thanks, that's very helpful (and also very prompt).  Much appreciated.

I'll see if the problem crops up under previous minor versions and update if I can narrow it down usefully.

**Comment by brazilofmux** [comment]
Don't use TinyMUX version 2.13.0.5 because it has CLI issues and data loss looms large with that, but I've been running stress clients against the server -- and fixing networking bugs. This maybe/might be getting at the issues you were seeing.
